libraries/fc/src/network/http/http_client.cpp

```
@@ -0,0 +1,361 @@
+      req.set(http::field::host, *dest.host());
+      req.set(http::field::user_agent, BOOST_BEAST_VERSION_STRING);
+      req.set(http::field::content_type, "application/json");
+      req.keep_alive(true);
+      req.body() = json::to_string(payload);
+      req.prepare_payload();
+
+      auto conn_iter = get_connection(dest, deadline);
+      auto eraser = make_scoped_exit([this, &conn_iter](){
+         _connections.erase(conn_iter);
+      });
+
+      // Send the HTTP request to the remote host
+      error_code ec = conn_iter->second.visit(write_request_visitor(this, req, deadline));
+      FC_ASSERT(!ec, "Failed to send request: ${message}", ("message",ec.message()));
+
+      // This buffer is used for reading and must be persisted
+      boost::beast::flat_buffer buffer;
+
+      // Declare a container to hold the response
+      http::response<http::string_body> res;
+
+      // Receive the HTTP response
+      ec = conn_iter->second.visit(read_response_visitor(this, buffer, res, deadline));
+      FC_ASSERT(!ec, "Failed to read response: ${message}", ("message",ec.message()));
+
+      // if the connection can be kept open, keep it open
+      if (res.keep_alive()) {
+         eraser.cancel();
+      }
+
+      auto result = json::from_string(res.body());
+      if (res.result() == http::status::internal_server_error) {
+         fc::exception_ptr excp;
+         try {
+            auto err_var = result.get_object()["error"].get_object();
+            excp = std::make_shared<fc::exception>(err_var["code"].as_int64(), err_var["name"].as_string(), err_var["what"].as_string());
+
+            if (err_var.contains("details")) {
+               auto details = err_var["details"].get_array();
+               for (const auto dvar : details) {
+                  excp->append_log(FC_LOG_MESSAGE(error, dvar.get_object()["message"].as_string()));
+               }
+            }
+         } catch( ... ) {
+
```

A log statement would be good here. If "error" is ever renamed to something else this will silently fail.

---

Nevermind, I see you FC_THROW below if this fails.